src/main/java/net/neoforged/neoforge/internal/CommonModLoader.java

```diff
@@ -41,7 +41,7 @@ public static boolean areRegistriesLoaded() {
         return registriesLoaded;
     }
 
-    protected static void begin(Runnable periodicTask) {
+    protected static void begin(Runnable periodicTask, boolean datagen) {
```

Comment:
I am not sure I like this architecture.

Have you considered creating a seperate `loadConfigs` method and invoking that from client and server, instead of altering this method and passing in a method?

Reply:
I have, and decided it would be more confusing short term than just adding a boolean. We can easily change it later if we have to make more tweaks to the startup phases.

Reply:
I understand your concern, but I rather we keep each new addition as simple as possible.

Would you mind changing it now already?

Reply:
Well, it's not as obvious as it might look. The `begin` method exists to contain everything that the client will do on the main thread; `load` is during the prepare phase of a reload listener; and `finish` is during the apply phase. So if we are moving the config loading call we probably want it in `begin`. But then we run into the issue with datagen, hence the check.

I expect that this class will change a number of times to address other issues, and until that happens it's not clear what the final architecture should be.

I'd see this extra `datagen` as a small red flag, i.e. something to keep an eye on when making further changes.

Reply:
Not convinced either. Calling it `datagen` makes it clear when reading `CommonModLoader` that loading the configs is explicitly not happening in datagen. All in all, this is not particularly important since this is internal implementation code that will change again in the future. Can we move forward with the PR? @shartte @marchermans

Reply:
Allthough I don't understand your rush: Yes we can move forward.